lib/ansible/modules/windows/win_auto_logon.ps1

```diff
@@ -9,75 +9,75 @@
 
 Set-StrictMode -Version 2.0
 
-$ErrorActionPreferrence = 'Stop'
-
-# argument spec hastable which has the input parameters and capabilities for this module
 $spec = @{
     options = @{
-        domain = @{type = "str"; default = $env:userdomain}
-        user = @{type = "str"; required = $true}
-        password = @{type = "str"; required = $true}
+        password = @{type = "str"; no_log = $true}
         state = @{type = "str"; choices = "absent","present"; default = "present"}
+        username = @{type = "str"}
     }
     required_if = @(
-        @("state", "present", @("user","password")),
-        @("state","absent",@())
+        , @("state", "present", @("username", "password"))
     )
 }
 
-$module = [Ansible.Basic.AnsibleModule]::Create($args,$spec)
-$domain = $module.params.domain
-$user = $module.params.user
+$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)
 $password = $module.params.password
 $state = $module.params.state
+$username = $module.params.username
+$domain = $null
 
-try {
-    #Build ParamHash
-
-    $autoAdminLogon = 1
-    $stateMessage = 'added'
-    if($state -eq 'absent'){
-        $autoadminlogon = 0
-        $stateMessage = 'removed'
-    }
-    $module.Result.Msg     = "Auto logon registry keys are already $stateMessage"
```

Comment:
Removed the `Msg` return value, this is typically reserved for just failure messages